geps/gep-2162.md

```diff
@@ -0,0 +1,112 @@
+# GEP-2162: Supported features in GatewayClass Status
+
+* Issue: [#2162](https://github.com/kubernetes-sigs/gateway-api/issues/2162)
+* Status: Provisional
+
+## TLDR
+
+This GEP proposes to enhance the [GatewayClassStatus](https://github.com/kubernetes-sigs/gateway-api/blob/f2cd9bb92b4ff392416c40d6148ff7f76b30e649/apis/v1beta1/gatewayclass_types.go#L185) to include a list of Gateway API features supported by the installed GatewayClass. 
+
+## Goals
+
+* Improve UX by enabling users to easily see what features the implementation(GatewayClass) support.
+
```

Comment:
```suggestion
* Provide foundation for tools to block or warn when unsupported features are used. 
```

Reply:
Or you are talking about gwctl?

Reply:
By "foundation" here, do you mean a library @robscott? That seems like it could be a good place to start, especially if we put the caveats about cross-resource validation into the docs for the library.

Reply:
I see foundation as "populating the data" so for future tools(gwctl etc)/webhook it will have the features populated in the gwc status. Ad those features are the foundation for the gwctl or the webhook which might use them.

one example I can think of could be `gwctl get-supported-features  --class=<gwc-name>`

The webhook example would be, say a user try to apply a HttpRoute manifest with features that is not supported by the class. So the ideas was that the webhook would block/or warn. BUT the cross-resource validation has a lot of caveats especially about eventual consistency. So I would vote to avoid it but thats a future discussion

Reply:
Done but leaving this thread open in case you disagree or have questions @youngnick

Reply:
Yeah, I think that wording is fine for now.